### ISSUE TITLE
Cannot generate schemas or encoders with external annotations

### ISSUE THREAD

**Issue by TJSomething** [sksamuel/avro4s#300]
I just upgraded to 2.0.4 from 1.8.3 and I am having issues with case classes that have or whose members have annotations for a different library.

This is probably the simplest example that doesn't work:

```scala
import com.sksamuel.avro4s._

@SerialVersionUID(1L)
final case class Foo(x: Int)

Encoder[Foo]
```

If I invoke the macro directly, this appears to caused by a failure to resolve the external annotation:

```
Error:(6, 29) exception during macro expansion:
scala.tools.reflect.ToolBoxError: reflective compilation has failed:
not found: value SerialVersionUID
not found: value value
at scala.tools.reflect.ToolBoxFactory$ToolBoxImpl$ToolBoxGlobal.throwIfErrors(ToolBoxFactory.scala:331)
at scala.tools.reflect.ToolBoxFactory$ToolBoxImpl$ToolBoxGlobal.wrapInPackageAndCompile(ToolBoxFactory.scala:213)
at scala.tools.reflect.ToolBoxFactory$ToolBoxImpl$ToolBoxGlobal.compile(ToolBoxFactory.scala:267)
at scala.tools.reflect.ToolBoxFactory$ToolBoxImpl.$anonfun$compile$13(ToolBoxFactory.scala:444)
at scala.tools.reflect.ToolBoxFactory$ToolBoxImpl$withCompilerApi$.apply(ToolBoxFactory.scala:370)
at scala.tools.reflect.ToolBoxFactory$ToolBoxImpl.compile(ToolBoxFactory.scala:437)
at com.sksamuel.avro4s.ReflectHelper.$anonfun$annotations$1(ReflectHelper.scala:157)
at scala.collection.immutable.List.map(List.scala:286)
at com.sksamuel.avro4s.ReflectHelper.annotations(ReflectHelper.scala:153)
at com.sksamuel.avro4s.NameResolution$.apply(NameResolution.scala:68)
at com.sksamuel.avro4s.SchemaFor$.applyMacroImpl(SchemaFor.scala:243)
SchemaFor.applyMacro[Foo]
```

**Comment by thesamet** [comment]
FWIW, this causes a ScalaPB interoperability issue. See https://stackoverflow.com/questions/56020490/derive-avro-schema-from-scala-case-class-generated-by-scalapb/56093071

**Comment by klass-ivan** [comment]
I think this one could be related - https://github.com/sksamuel/avro4s/issues/302

**Comment by sksamuel** [comment]
Can you try the 3.0.x snapshot from 
https://oss.sonatype.org/content/repositories/snapshots/com/sksamuel/avro4s/avro4s-core_2.12/
This should be fixed and is available for 2.11 and 2.22

**Comment by TJSomething** [comment]
Sorry I didn't get to this sooner. From a quick test in Ammonite, it looks like this is fixed.
